internal/handlers/lokistack_create_or_update.go

```diff
@@ -61,6 +63,40 @@ func CreateOrUpdateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client
 		)
 	}
 
+	var tenantSecrets []*manifests.TenantSecrets
+	if stack.Spec.Tenants != nil {
+		if err = gateway.ValidateModes(stack); err != nil {
+			return kverrors.Wrap(err, "invalid configuration provided for given mode")
+		}
```

Comment:
Throwing the error into the operator log file will usually go unnoticed and is bad user experience. This should set the LokiStackCR into a custom degraded condition, e.g.:
```go
return status.SetDegradedCondition(ctx, k, req,
    fmt.Sprintf("Invalid tenants configuration: %s", err),
    lokiv1beta1.ReasonInvalidTenantsConfiguration,
)
```